packages/next/src/shared/lib/router/router.ts

```diff
@@ -1323,7 +1323,7 @@ export default class Router implements BaseRouter {
     // if we detected the path as app route during prefetching
     // trigger hard navigation
     if ((this.components[pathname] as any)?.__appRouter) {
-      handleHardNavigation({ url, router: this })
+      handleHardNavigation({ url: as, router: this })
```

Comment:
`as` seems to be used for next 3 instances of `handleHardNavigation`

Line 1343:

https://github.com/chibicode/next.js/blob/749a839a632326f0cae29ec8f6b3446f3ae662b9/packages/next/src/shared/lib/router/router.ts#L1343

Line 1405:

https://github.com/chibicode/next.js/blob/749a839a632326f0cae29ec8f6b3446f3ae662b9/packages/next/src/shared/lib/router/router.ts#L1405

Line 1443:

https://github.com/chibicode/next.js/blob/749a839a632326f0cae29ec8f6b3446f3ae662b9/packages/next/src/shared/lib/router/router.ts#L1443